tests/test-class-amp-story-post-type.php

```diff
@@ -40,6 +40,54 @@ public function test_the_single_story_card() {
 		}
 	}
 
+	/**
+	 * Test get_embed_template.
+	 *
+	 * @covers AMP_Story_Post_Type::get_embed_template()
+	 */
+	public function test_get_embed_template() {
+		$template          = '/srv/www/baz.php';
+		$wrong_type        = 'post';
+		$correct_type      = 'embed';
+		$wrong_templates   = array( 'embed-testimonal.php', 'embed.php' );
+		$correct_template  = sprintf( 'embed-%s.php', AMP_Story_Post_Type::POST_TYPE_SLUG );
+		$expected_template = 'embed-amp-story.php';
+		$correct_templates = array( $correct_template, 'embed.php' );
+
+		$this->assertEquals( $template, AMP_Story_Post_Type::get_embed_template( $template, $wrong_type, $correct_templates ) );
+		$this->assertEquals( $template, AMP_Story_Post_Type::get_embed_template( $template, $correct_type, $wrong_templates ) );
+		$this->assertContains( $expected_template, AMP_Story_Post_Type::get_embed_template( $template, $correct_type, $correct_templates ) );
+	}
+
+	/**
+	 * Test enqueue_embed_styling.
+	 *
+	 * @covers AMP_Story_Post_Type::enqueue_embed_styling()
+	 */
+	public function test_enqueue_embed_styling() {
+		AMP_Story_Post_Type::enqueue_embed_styling();
+		// None of the conditional is satisfied, so this should not enqueue the stylesheet.
+		$this->assertFalse( wp_style_is( AMP_Story_Post_Type::STORY_CARD_CSS_SLUG ) );
+
+		// Only the first part of the conditional is satisfied, so this again should not enqueue the stylesheet.
+		$GLOBALS['wp_query']->is_embed = true;
+		AMP_Story_Post_Type::enqueue_embed_styling();
+		$this->assertFalse( wp_style_is( AMP_Story_Post_Type::STORY_CARD_CSS_SLUG ) );
+
+		$GLOBALS['wp_query']->is_embed       = true;
```

Comment:
Instead of manually setting the query here, I'd try to use `$this->go_to()`, which is a neat little helper method provided by the WordPress core test suite. I think it should work well for exactly this type of test.

Reply:
Sure, https://github.com/ampproject/amp-wp/pull/1944/files#diff-694f9f17a2563b4b06a20b719fa41466R92 now uses `$this->go_to()` instead of changing `$GLOBALS['wp_query']`.